cli/server/server.go

```diff
@@ -302,12 +315,12 @@ func startServer(ctx *cli.Context) error {
 
 	serverConfig := network.NewServerConfig(cfg)
 
-	chain, prometheus, pprof, err := initBCWithMetrics(cfg)
+	chain, prometheus, pprof, err := initBCWithMetrics(cfg, log)
 	if err != nil {
 		return err
 	}
 
-	server := network.NewServer(serverConfig, chain)
+	server := network.NewServer(serverConfig, chain, nil)
```

Comment:
`nil` logger for a server? I don't think it's a good idea. We should have the same logging level after this changeset as we had before (not counting consensus debugging, of course, that should go away).

Reply:
Also note that we should respect users debug choice for all packages at the moment (I guess we'll need to implement per-package logging choices in the future, but not now).

Reply:
oops, it is a mistake probably, `nil` was intended only for tests.